File: TopInterview150/125_valid_palindrome/isPalindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        if (s.isEmpty()) {
            return true;
        }

        int left = 0;
        int right = s.length() - 1;

        while (left < right) {
            char charLeft = s.charAt(left);
            char charRight = s.charAt(right);

            if (!Character.isLetterOrDigit(charLeft)) {
                left++;
            } else if (!Character.isLetterOrDigit(charRight)) {
                right--;
            } else {
                if (Character.toLowerCase(charLeft) != Character.toLowerCase(charRight)) {
                    return false;
                }
                left++;
                right--;
            }
        }

        return true;
    }
}

File: 125_valid_palindrome/isPalindrome.ts
function isPalindrome(s: string): boolean {
  if (s.trim() === "") {
    return true;
  }

  let str = "";

  const lowerString = s.toLowerCase();

  for (let i = 0; i <= lowerString.length - 1; i++) {
    if (
      (lowerString[i] >= "a" && lowerString[i] <= "z") ||
      (lowerString[i] >= "0" && lowerString[i] <= "9")
    ) {
      str += lowerString[i];
    }
  }

  let left = 0;
  let right = str.length - 1;

  while (left < right) {
    if (str.charCodeAt(left) === str.charCodeAt(right)) {
      left++;
      right--;
    } else {
      return false;
    }
  }

  return true;
}
